File: src/com/eagle/programmar/Bash/Commands/Bash_ReadCommand.java
// Copyright Eagle Legacy Modernization, LLC, 2010-date
// Original author: Steven A. O'Hara, Jul 20, 2022

package com.eagle.programmar.Bash.Commands;

import com.eagle.programmar.Bash.Symbols.Bash_Identifier_Reference;
import com.eagle.programmar.Bash.Terminals.Bash_Keyword;
import com.eagle.programmar.Bash.Terminals.Bash_KeywordChoice;
import com.eagle.programmar.Bash.Terminals.Bash_Literal;
import com.eagle.tokens.TokenChooser;
import com.eagle.tokens.TokenList;
import com.eagle.tokens.TokenSequence;
import com.eagle.tokens.interfaces.AbstractStatement;

public class Bash_ReadCommand extends TokenSequence implements AbstractStatement
{
	public @S(10) Bash_Keyword READ = new Bash_Keyword("read");
	public @S(20) @OPT TokenList<Bash_ReadOption> options;
	public @S(30) TokenList<Bash_Identifier_Reference> ids;

	public static class Bash_ReadOption extends TokenChooser
	{
		public @CHOICE static class Bash_ReadPrompt extends TokenSequence
		{
			public @S(10) Bash_KeywordChoice opt = new Bash_KeywordChoice("-p");
			public @S(20) Bash_Literal prompt;
		}
	}
}

File: csrc/com/eagle/programmar/Bash/Commands/Bash_ReadCommand.cs
// ====================================================================================================
// Produced by the Free Edition of Java to C# Converter.
// Purchase a Premium Edition license at:
// https://www.tangiblesoftwaresolutions.com/order/order-java-to-csharp.html
// ====================================================================================================

// Copyright Eagle Legacy Modernization, LLC, 2010-date
// Original author: Steven A. O'Hara, Jul 20, 2022

namespace com.eagle.programmar.Bash.Commands
{
	using Bash_Identifier_Reference = com.eagle.programmar.Bash.Symbols.Bash_Identifier_Reference;
	using Bash_Keyword = com.eagle.programmar.Bash.Terminals.Bash_Keyword;
	using Bash_KeywordChoice = com.eagle.programmar.Bash.Terminals.Bash_KeywordChoice;
	using Bash_Literal = com.eagle.programmar.Bash.Terminals.Bash_Literal;
	using TokenChooser = com.eagle.tokens.TokenChooser;
	using TokenList = com.eagle.tokens.TokenList;
	using TokenSequence = com.eagle.tokens.TokenSequence;
	using AbstractStatement = com.eagle.tokens.interfaces.AbstractStatement;

	public class Bash_ReadCommand : TokenSequence, AbstractStatement
	{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(10) com.eagle.programmar.Bash.Terminals.Bash_Keyword READ = new com.eagle.programmar.Bash.Terminals.Bash_Keyword("read");
		public Bash_Keyword READ = new Bash_Keyword("read");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(20) @OPT TokenList<Bash_ReadOption> options;
		public  OPT;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(30) com.eagle.tokens.TokenList<com.eagle.programmar.Bash.Symbols.Bash_Identifier_Reference> ids;
		public TokenList<Bash_Identifier_Reference> ids;

		public class Bash_ReadOption : TokenChooser
		{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @CHOICE static class Bash_ReadPrompt extends com.eagle.tokens.TokenSequence
			public class Bash_ReadPrompt : TokenSequence
			{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(10) com.eagle.programmar.Bash.Terminals.Bash_KeywordChoice opt = new com.eagle.programmar.Bash.Terminals.Bash_KeywordChoice("-p");
				public Bash_KeywordChoice opt = new Bash_KeywordChoice("-p");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(20) com.eagle.programmar.Bash.Terminals.Bash_Literal prompt;
				public Bash_Literal prompt;
			}
		}
	}

}
